Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm

```diff
@@ -1585,6 +1585,56 @@ static void checkColor(WebCore::CocoaColor *color, std::optional<ColorExpectatio
     TestWebKitAPI::Util::run(&finished);
 }
 
+TEST(WritingTools, CompositionShowOriginalHasNoTransparentMarkers)
+{
+    RetainPtr session = adoptNS([[WTSession alloc] initWithType:WTSessionTypeComposition textViewDelegate:nil]);
+
+    RetainPtr webView = adoptNS([[WritingToolsWKWebView alloc] initWithHTMLString:@"<body id='p' contenteditable><div id='first'><b>Dream interpretation is a skill that anyone can learn</b>. It’s like a treasure hunt into our subconscious minds.</div></body>"]);
+    [webView focusDocumentBodyAndSelectAll];
+
+    RetainPtr originalText = @"Dream interpretation is a skill that anyone can learn. It’s like a treasure hunt into our subconscious minds.";
+
+    __block bool finished = false;
+
+    [[webView writingToolsDelegate] willBeginWritingToolsSession:session.get() requestContexts:^(NSArray<WTContext *> *contexts) {
+        EXPECT_EQ(1UL, contexts.count);
+
+        EXPECT_WK_STREQ(originalText.get(), contexts.firstObject.attributedText.string);
+
+        [[webView writingToolsDelegate] writingToolsSession:session.get() didReceiveAction:WTActionCompositionRestart];
+
+        EXPECT_WK_STREQ(originalText.get(), [webView contentsAsStringWithoutNBSP]);
+
+        RetainPtr fontManager = [NSFontManager sharedFontManager];
+        RetainPtr boldFont = [fontManager fontWithFamily:@"Helvetica" traits:NSBoldFontMask weight:0 size:14];
```

Comment:
I don't think this will build on iOS?

Reply:
It won't, but also this test is guarded behind `PLATFORM(MAC)`.

Reply:
It certainly wasn't in the version of the patch I reviewed 😛

Reply:
But also, why is the test Mac only now?